### PR TITLE
Move viewing service into the core folder and update appearance of viewing component

### PR DESCRIPTION
**Pull request by LillianHuang-dev**
Changes by @TommyHuang-dev:
- Move viewing service into the core folder for consistency since other components may need to use it.

Changes by @janakitti:
- Animate response-list button with the rest of the list when opening/closing it on desktop
- Reposition response-list button on mobile for easier one-handed access
- Implement a new-to-viewing cookie to activate an animation that draws attention to the response-list button
- Reword the bookmark toast

### CODE REVIEW

**Review by janakitti — APPROVED** [review]

File: src/app/pages/viewing/responses-list/responses-list.component.ts [hunk 1]
```diff
@@ -43,22 +47,39 @@ export class ResponsesListComponent implements OnInit {
...
+    }
+
+    this.showNewToViewingValue = this.cookieService.get('viewing-cookie');
+
+    if (this.showNewToViewingValue === 'true') {
+      this.showNewToViewing = true;
+    } else if (this.showNewToViewingValue === 'false') {
+      this.showNewToViewing = false;
+    } else {
+      console.log('Error. Viewing-cookie not found or initialized properly.');
```
Comment (DabeerAzeez) [inline src/app/pages/viewing/responses-list/responses-list.component.ts]: Why do we have a `console.log()` here?

**Review by DabeerAzeez — CHANGES_REQUESTED** [review]

**Comment by DabeerAzeez** [comment]
@TommyHuang-dev there seems to be a lot of other changes on this PR other than just moving viewing component into `/core`? This PR needs an accurate description of everything that's been changed to be placed inside the top comment for changelog and other purposes. Were these changes that were already there before and you're just moving them in or something?

**Review by jonapap — APPROVED** [review]

**Comment by LillianHuang-dev** [comment]
> @TommyHuang-dev there seems to be a lot of other changes on this PR other than just moving viewing component into `/core`? This PR needs an accurate description of everything that's been changed to be placed inside the top comment for changelog and other purposes. Were these changes that were already there before and you're just moving them in or something?

@GuyInFridge 

It's because commits were pushed after the PR was opened. I edited my initial description to reflect the new changes.

File: src/app/pages/viewing/responses-list/responses-list.component.css [hunk 2]
```diff
@@ -21,51 +27,98 @@
   margin-left: 50px;
 }
 
-.close-btn {
+/* Button component */
+.hamburger {
+  position: absolute;
+  height: 6em;
+  width: 2em;
+  top: calc(2em);
```
Comment (ricetech) [inline src/app/pages/viewing/responses-list/responses-list.component.css]: Why is there a calc without any variables?
```suggestion
  top: 2em;
```

**Review by ricetech — CHANGES_REQUESTED** [review]

**Comment by ricetech** [comment]
uhhhh
image

**Review by ricetech — CHANGES_REQUESTED** [review]
@janakitti 
image

File: src/app/pages/viewing/responses-list/responses-list.component.ts (hunk 1 above)
Comment (janakitti) [reply src/app/pages/viewing/responses-list/responses-list.component.ts]: You had that in your New-Cookie condition in app-module. I just kept it to be consistent, but I guess it's not needed

Comment (ricetech) [reply src/app/pages/viewing/responses-list/responses-list.component.ts]: Err, couldn't you just initialize a new cookie there if there's an error? In any case, we should not be using console.log at all, instead handling the error or using an Alert.

**Review by ricetech — CHANGES_REQUESTED** [review]
Can the mobile chevron also move with the list? Also I would change the opacity during the 'draw attention' animation to be more opaque

File: src/app/pages/viewing/responses-list/responses-list.component.ts (hunk 1 above)
Comment (DabeerAzeez) [reply src/app/pages/viewing/responses-list/responses-list.component.ts]: @janakitti that was a mistake lmao, I changed it to `AuthService.displayError()`

Comment (DabeerAzeez) [inline src/app/pages/viewing/responses-list/responses-list.component.ts]:
```suggestion
     AuthService.displayError('Viewing-cookie not found or initialized properly.');
```

Comment (DabeerAzeez) [reply src/app/pages/viewing/responses-list/responses-list.component.ts]: @janakitti  / @TommyHuang-dev you'll have to double check if this works, I edited this in from the web editor.

**Review by janakitti — APPROVED** [review]

**Review by anthonyftwang — APPROVED** [review]

**Review by ricetech — APPROVED** [review]

**Review by DabeerAzeez — APPROVED** [review]

**Pull request merged**